components/proposal/Card.vue

```diff
@@ -198,27 +204,37 @@ const emit = defineEmits<{
 }>();
 
 const apiStore = useApiClientStore();
-
+const localSelectedVotes = useLocalSelectedVotes();
 const { address: userAccount, isConnected, isDisconnected } = useAccount();
 
-const selectedVote = ref<null | boolean>(null);
-const reasonForVoteCheckbox = ref<boolean | undefined>(false);
-const reasonForVote = ref<string>("");
+const localStoredVote = computed(() =>
+  localSelectedVotes.get(props.proposal.proposalId),
+);
+
+const selectedVote = computed(() =>
+  localStoredVote.value != null
+    ? localStoredVote.value.vote === 0
+      ? false
+      : true
+    : null,
+);
+
+const reasonForVoteCheckbox = ref<boolean | undefined>(
+  Boolean(localStoredVote.value?.reason),
+);
 const reasonForVoteTextarea = ref();
 
 watch(reasonForVoteCheckbox, async (value) => {
   if (value) {
+    // Focus the textarea when enabled
     await nextTick();
-    reasonForVoteTextarea?.value.focus();
+    reasonForVoteTextarea.value?.focus();
```

Comment:
btw, this doesn't really work. spent some time on it with no luck. textarea doesn't focus even if mounted already.